### ISSUE TITLE
Add an option to prevent double submission of forms

### ISSUE THREAD

**Issue by hamidpeywasti** [contao/contao#1933]
Currently users can submit a form more than once by clicking "Submit button" (in comments or forms) continuously and fast. Hence there was problem with multiple request from a form (i.e. 3 same comments at same time ). So I figure out that it happens in mobile Phones (maybe touch screen!) not desktop.

My instant solution was adding a javascript code to submit button:

`onclick="this.disabled=true;this.value='Sending, please wait...';this.form.submit();"`

But in this case, if there was an error in form, user cannot anymore submit form because the submit button is disabled. So "Disable HTML5 validation" must be checked in form setting. This worked for me, But there must be a good universal solution.

**Comment by hamidpeywasti** [comment]
This code simply can prevent submitting a form twice and work for all Contao forms with HTML5 validation.

```javascript
<script>
$(function() {
    // prevent the submit button to be pressed twice
    $("form").submit(function() {
        $(this).find('.submit').attr('disabled', true);
        $(this).find('.submit').text('Sending...');
    });
})
</script>
```

**Comment by ausi** [comment]
I don’t think we should add such a prevention by default. Double submition has use cases and in cases where it creates an actual problem it has to be prevented on the server side anyways.

See also https://github.com/whatwg/html/issues/5312

**Comment by fritzmg** [comment]
> in cases where it creates an actual problem it has to be prevented on the server side anyways

Yeah but isn't that something the `Form` class could (optionally) do in that case?

**Comment by leofeyer** [comment]
As discussed in Mumble on July 30th, the double submission should be prevented server-side and it should be disabled by default.

**Comment by zoglo** [comment]
Surely it's on client-side right now (see #7293) but can this be closed?

**Comment by fritzmg** [comment]
I think we can close this as a client-side solution is now built into Contao 5.3.10+ via the PR mentioned above (https://github.com/contao/contao/pull/7293).
